src/components/binding/binding-renderer.js

```diff
@@ -96,45 +102,53 @@ genotet.BindingRenderer = function(container, data) {
-/** @const {number} */
-genotet.BindingRenderer.prototype.EXON_CENTER_Y = 20;
-/** @const {number} */
-genotet.BindingRenderer.prototype.EXON_LABEL_OFFSET = 3;
-/** @const {number} */
-genotet.BindingRenderer.prototype.OVERVIEW_HEIGHT = 25;
-/** @const {number} */
-genotet.BindingRenderer.prototype.BED_HEIGHT = 25;
-/** @const {number} */
-genotet.BindingRenderer.prototype.BED_SHALLOW_WIDTH = 1;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.EXON_HEIGHT_ = 35;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.EXON_SIZE_ = 10;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.EXON_CENTER_Y_ = 20;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.EXON_LABEL_OFFSET_ = 3;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.OVERVIEW_HEIGHT_ = 25;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.BED_HEIGHT_ = 25;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.BED_SHALLOW_WIDTH_ = 1;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.TRACK_NAME_HEIGHT_ = 50;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.TRACK_NAME_TEXT_OFFSET_ = 10;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.TRACK_NAME_TEXT_HEIGHT_ = 10;
 
 /**
  * When there are more than this limit number of exons, we draw exons as
  * abstract rectangles.
- * @type {number}
+ * @private {number}
  */
-genotet.BindingRenderer.prototype.EXON_ABSTRACT_LIMIT = 200;
-
-/** @const {number} */
-genotet.BindingRenderer.prototype.STRAND_HORIZONTAL_SIZE = 5;
-/** @const {number} */
-genotet.BindingRenderer.prototype.STRAND_VERTICAL_SIZE = 3;
-/** @const {number} */
-genotet.BindingRenderer.prototype.EXON_LABEL_SIZE = 6;
-/** @const {number} */
-genotet.BindingRenderer.prototype.BED_LABEL_SIZE = 13;
-/** @const {number} */
-genotet.BindingRenderer.prototype.BED_MIN_WIDTH = 3;
-/** @const {number} */
-genotet.BindingRenderer.prototype.BED_HEIGHT_PROPORTION = 0.8;
-
-/** @const {!Array<number>} */
-genotet.BindingRenderer.ZOOM_EXTENT = [1, 20000000];
-/** @const {number} */
-genotet.BindingRenderer.ZOOM_INTERVAL = 500;
+genotet.BindingRenderer.prototype.EXON_ABSTRACT_LIMIT_ = 200;
+
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.STRAND_HORIZONTAL_SIZE_ = 5;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.STRAND_VERTICAL_SIZE_ = 3;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.EXON_LABEL_SIZE_ = 6;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.BED_LABEL_SIZE_ = 13;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.BED_MIN_WIDTH_ = 3;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.BED_MIN_HEIGHT_ = 3;
+/** @private @const {number} */
+genotet.BindingRenderer.prototype.BED_HEIGHT_EXTENSION_ = 200;
+
+/** @private @const {!Array<number>} */
+genotet.BindingRenderer.ZOOM_EXTENT_ = [1, 20000000];
```

Comment:
Otherwise you would get compiler error saying It cannot have non-primitive
type as zoom extent is an array.

On Saturday, February 20, 2016, Liana Yang <notifications@github.com> wrote:

> In src/components/binding/binding-renderer.js
> <https://github.com/ViDA-NYU/genotet/pull/115#discussion_r53557655>:
>
> > +genotet.BindingRenderer.prototype.STRAND_HORIZONTAL_SIZE_ = 5;
> > +/** @private @const {number} */
> > +genotet.BindingRenderer.prototype.STRAND_VERTICAL_SIZE_ = 3;
> > +/** @private @const {number} */
> > +genotet.BindingRenderer.prototype.EXON_LABEL_SIZE_ = 6;
> > +/** @private @const {number} */
> > +genotet.BindingRenderer.prototype.BED_LABEL_SIZE_ = 13;
> > +/** @private @const {number} */
> > +genotet.BindingRenderer.prototype.BED_MIN_WIDTH_ = 3;
> > +/** @private @const {number} */
> > +genotet.BindingRenderer.prototype.BED_MIN_HEIGHT_ = 3;
> > +/** @private @const {number} */
> > +genotet.BindingRenderer.prototype.BED_HEIGHT_EXTENSION_ = 200;
> > +
> > +/** @private @const {!Array<number>} */
> > +genotet.BindingRenderer.ZOOM_EXTENT_ = [1, 20000000];
>
> why using function return?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/115/files#r53557655>.
>